[Marketplace Contribution] Tenable.sc - Content Pack Update (#35341)
demisto/content: Packs/Tenable_sc/Integrations/Tenable_sc/README.md

#### Context Output

| **Path** | **Type** | **Description** |
### tenable-sc-list-repositories

***

> Requires security manager role. Get a list of Tenable.sc scan repositories.

Reviewer: ```suggestion
Get a list of Tenable.sc scan repositories. Requires security manager role. 
```

| auth_type | The authentication type. Tenable (TNS). Lightweight Directory Access Protocol (LDAP). Security Assertion Markup Language (SAML). LDAP server or SAML authentication needs to be configured in order to select LDAP or SAML. Possible values are: ldap, legacy, linked, saml, tns. Default is tns. | Required | 
| password | The user's password. Must be at least 3 characters. | Required | 
| time_zone | The user timezone, possible values can be found here: https://docs.oracle.com/middleware/1221/wcs/tag-ref/MISC/TimeZones.html. | Optional | 
| role_id | The user's role. Only an Administrator can create Administrator accounts. Possible values are: Administrator, Security Manager, Security Analyst, Vulnerability Analyst, Executive, Credential Manager, Auditor. | Required | 
| must_change_password | Whether the password must be changed. When choosing LDAP or SAML auth types, 'must_change_password' must be set to False. For all other cases can be either True or False. Possible values are: false, true. Default is false. | Optional | 

> | managed_users_groups | Comma-separated list of session user's role that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 
> | managed_objects_groups | Comma-separated list of the session user's role that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 

Reviewer: ```suggestion
| managed_users_groups | Comma-separated list of session user's roles that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 
| managed_objects_groups | Comma-separated list of the session user's roles that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 
```

| group_id | Valid group ID whose users can be managed by the created user. Default is 0. | Required | 

> | responsible_asset_id | Default is 0. ID of a valid, usable, accessible asset. Use tenable-sc-list-assets to get all available assets. -1 is not set, 0 is all assets, and other numbers are asset ID. Default is 0. | Required | 

Reviewer: ```suggestion
| responsible_asset_id | ID of a valid, usable, accessible asset. Use tenable-sc-list-assets to get all available assets. -1 is not set, 0 is all assets, and other numbers are asset ID. Default is 0. | Required | 
```

| time_zone | The user timezone. Possible values can be found here: https://docs.oracle.com/middleware/1221/wcs/tag-ref/MISC/TimeZones.html. | Optional | 
| role_id | The user's role. Only an Administrator can create Administrator accounts. Possible values are: Administrator, Security Manager, Security Analyst, Vulnerability Analyst, Executive, Credential Manager, Auditor. | Optional | 
| must_change_password | Whether the password must be changed. When choosing LDAP or SAML auth types, 'must_change_password' must be set to False. For all other cases can be either True or False. Possible values are: false, true. Default is false. | Optional | 
| managed_users_groups | Comma-separated list of session user's role that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 

> | managed_objects_groups | Comma-separated list of session user's role that  can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 

Reviewer: ```suggestion
| managed_objects_groups | Comma-separated list of session user's role that can manage groups. Use tenable-sc-list-groups to get all available groups. Default is 0. | Optional | 
```

#### Context Output

There is no context output for this command.
| family_id | Family ID. Can be retrieved from the result of the tenable-sc-list-plugin-family command. | Required | 

> | plugins_id | Comma-separated list of plugin_ids, Can be retrieved from the result of  the tenable-sc-list-plugin-family command  with family_id as the argument. | Required | 

Reviewer: ```suggestion
| plugins_id | Comma-separated list of plugin_ids. Can be retrieved from the result of  the tenable-sc-list-plugin-family command  with family_id as the argument. | Required | 
```

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| TenableSC.Scan.Assets | Unknown | Scan assets. | 
| TenableSC.Scan.CanManage | String | Scan permissions. | 
| TenableSC.Scan.CanUse | String | Scan permissions. | 
| TenableSC.Scan.ClassifyMitigatedAge | String | Scan classify mitigated age. | 
| TenableSC.Scan.CreatedTime | Date | Scan creation time. | 
| TenableSC.Scan.Creator.Firstname | String | Scan creator first name. | 
### tenable-sc-update-asset

***
Requires security manager role. Update an asset.

#### Base Command

`tenable-sc-update-asset`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| name | Asset name. | Optional | 
| asset_id | The ID of the asset to update. | Required | 
| description | The asset description. | Optional | 
| owner_id | The asset owner ID. | Optional | 
| tag | The asset tag. | Optional | 

> | ip_list | Comma-separated list of the asset IPs list. | Optional | 

Reviewer: ```suggestion
| ip_list | Comma-separated list of the asset IPs. | Optional | 
```

#### Context Output

There is no context output for this command.
| TenableSC.Scan.Policy.Description | String | Scan policy description. | 
| TenableSC.Scan.Policy.ID | String | Scan policy ID. | 
| TenableSC.Scan.Policy.Name | String | Scan policy name. | 
| TenableSC.Scan.Policy.Owner.Firstname | String | Scan policy owner first name. | 
| TenableSC.Scan.Policy.Owner.ID | String | Scan policy owner ID. | 
### tenable-sc-create-remediation-scan

***
Creates a remediation scan. Requires security manager role. This command is a prerequisite for creating remediation scan.

#### Base Command

`tenable-sc-create-remediation-scan`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| policy_name | The name of the policy to create. | Optional | 
| policy_description | The description of the policy to create. | Optional | 

> | port_scan_range | Possible values: default, all or a comma-separated list of values - 21,23,25,80,110. | Optional | 

Reviewer: ```suggestion
| port_scan_range | 'The port scan range. Possible values: default, all, or a comma-separated list of values - 21,23,25,80,110.' | Optional | 
```

> | tcp_scanner | Only possible if you are using Linux or FreeBSD. On Windows or macOS, the scanner does not do a TCP scan and instead uses the SYN scanner..If you enable this option, you can also set the syn_firewall_detection. Possible values are: no, yes. Default is no. | Optional | 

Reviewer: ```suggestion
| tcp_scanner | Only possible if you are using Linux or FreeBSD. On Windows or macOS, the scanner does not do a TCP scan and instead uses the SYN scanner. If you enable this option, you can also set the syn_firewall_detection. Possible values are: no, yes. Default is no. | Optional | 
```

| syn_scanner | Identifies open TCP ports on the target hosts. If you enable this option, you can also set the syn_firewall_detection option. Possible values are: no, yes. Default is yes. | Optional | 

> | udp_scanner | Enabling the UDP port scanner may dramatically increase the scan time and produce unreliable results. Consider using the netstat or SNMP port enumeration options instead if possible. Possible values are: no, yes. Default is no. | Optional | 

Reviewer: ```suggestion
| udp_scanner | Whether to enable the UDP port scanner. Enabling the UDP port scanner may dramatically increase the scan time and produce unreliable results. Consider using the netstat or SNMP port enumeration options instead if possible. Possible values are: no, yes. Default is no. | Optional | 
```

> | syn_firewall_detection | Default is Automatic (normal). Rely on local port enumeration first before relying on network port scans. Possible values are: Automatic (normal), Do not detect RST rate limitation(soft), Ignore closed ports(aggressive), Disabled(softer). Default is Automatic (normal). | Optional | 

Reviewer: ```suggestion
| syn_firewall_detection | Rely on local port enumeration first before relying on network port scans. Possible values are: Automatic (normal), Do not detect RST rate limitation(soft), Ignore closed ports(aggressive), Disabled(softer). Default is Automatic (normal). | Optional | 
```

> | family_id | Can be retrieved from the result of the tenable-sc-list-plugin-family command. | Required | 

Reviewer: ```suggestion
| family_id | The family ID. Can be retrieved from the result of the tenable-sc-list-plugin-family command. | Required | 
```

> | plugins_id | Comma-separated list of plugin_ids, Can be retrieved from the result of the tenable-sc-list-plugin-family command  with family_id as the argument. | Required | 

Reviewer: ```suggestion
| plugins_id | Comma-separated list of plugin_ids, Can be retrieved from the result of the tenable-sc-list-plugin-family command with family_id as the argument. | Required | 
```

| scan_name | Scan name. | Required | 
| description | Scan description. | Optional | 

> | repository_id | Scan Repository ID, can be retrieved from the list-repositories command. Default is 1. | Required | 

Reviewer: ```suggestion
| repository_id | Scan repository ID. Ccan be retrieved from the list-repositories command. Default is 1. | Required | 
```

| time_zone | The timezone for the given start_time. Possible values can be found here: https://docs.oracle.com/middleware/1221/wcs/tag-ref/MISC/TimeZones.html. | Optional | 
| start_time | The scan start time, in the format of YYYY-MM-DD:HH:MM:SS or relative timestamp (i.e., now, 3 days). | Optional | 
| repeat_rule_freq | Specifies repeating events based on an interval of a repeat_rule_freq or more. Possible values are: HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY. | Optional | 
| repeat_rule_interval | The number of repeat_rule_freq between each interval (for example: If repeat_rule_freq=DAILY and repeat_rule_interval=8 it means every eight days.). | Optional | 
| repeat_rule_by_day | A comma-separated list of days of the week to run the schedule. Possible values are: SU, MO, TU, WE, TH, FR, SA. | Optional | 
| asset_ids | Either no assets or comma-separated list of asset IDs to scan. Can be retrieved from the list-assets command. | Optional | 

> | scan_virtual_hosts | Default is false. Whether to include virtual hosts. Possible values are: true, false. Default is false. | Optional | 

Reviewer: ```suggestion
| scan_virtual_hosts | Whether to include virtual hosts. Possible values are: true, false. Default is false. | Optional | 
```

> | ip_list | Comma-separated IPs to scan, e.g., 10.0.0.1,10.0.0.2 . | Optional | 

Reviewer: ```suggestion
| ip_list | Comma-separated list of IPs to scan, e.g., 10.0.0.1,10.0.0.2 . | Optional | 
```

| report_ids | Comma-separated list of report definition IDs to create post-scan. Can be retrieved from the list-report-definitions command. | Optional | 
| credentials | Comma-separated credentials IDs to use. Can be retrieved from the list-credentials command. | Optional | 
| timeout_action | discard - do not import any of the results obtained by the scan to the database. import - Import the results of the current scan and discard the information for any unscanned targets. rollover-Import the results from the scan into the database and create a rollover scan that may be launched at a later time to complete the scan. Possible values are: discard, import, rollover. Default is import. | Optional | 
| max_scan_time | Maximum scan run time in hours. Default is 1. | Optional | 

> | dhcp_tracking | Track hosts which have been issued new IP address, (e.g., DHCP). Possible values are: true, false. Default is false. | Optional | 

Reviewer: ```suggestion
| dhcp_tracking | Whether to track hosts that have been issued new IP address, (e.g., DHCP). Possible values are: true, false. Default is false. | Optional | 
```

| enabled | Whether the schedule is enabled. The "enabled" field can only be set to "false" for schedules of type "ical". For all other schedules types, "enabled" is set to "true". Possible values are: true, false. Default is true. | Optional | 

> | rollover_type | Create a rollover scan scheduled to launch the next day at the same start time as the just completed scan. template-Create a rollover scan as a template for users to launch manually This field is required if the timeout_action is set to rollover. Default is nextDay. | Optional | 

Reviewer: ```suggestion
| rollover_type | Create a rollover scan scheduled to launch the next day at the same start time as the just completed scan. template-Create a rollover scan as a template for users to launch manually. This field is required if the timeout_action is set to rollover. Default is nextDay. | Optional | 
```
